### ISSUE TITLE
Properly uniquely identify TLS client certificate subjects and issuers

### ISSUE THREAD

**Issue by tpdownes** [containous/traefik#4199]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

That Traefik would communicate all the information necessary for frontends to compare certificate Distinguished Names ("subjects") to those it has stored for access control.

Traefik does not presently support all the [subject fields required](https://tools.ietf.org/html/rfc3739#section-3.1.2) and [any issuer information whatsoever](https://tools.ietf.org/html/rfc3739#section-3.1.1) to [uniquely identify](https://tools.ietf.org/html/rfc3739#section-2.4) a certificate.

That is to say, you do not properly reconstruct certificate Distinguished Names and they are only unique for a given issuer. If you were to use a bundle to verify certificates (e.g., on Debian, `/etc/ssl/certs/ca-certificates.crt`) you do not provide the information necessary for the backend to double-check the issuer.

As a practical matter, I will have to pass the whole PEM and check its subject on the frontend. If I'm doing that, I might as well be verifying it myself.

Here's my Subject DN:
```
Subject: DC=org, DC=cilogon, C=US, O=LIGO, CN=Thomas Downes thomas.downes@ligo.org
```
Here's my Issuer DN:
```
Issuer: DC=org, DC=cilogon, C=US, O=CILogon, CN=CILogon Basic CA 1
```
In particular, Traefik doesn't know about `DC`, only `O`. The RFC must specify how to uniquely re-construct these DNs (UPDATE: well, it doesn't, but it does have specified rules about comparison which Traefik can't implement. Though, I believe OpenSSL has a "canonical" approach to constructing the DN) and you should really be offering that as an option. Something like:
```
traefik.frontend.passTLSClientCert.subject=true
traefik.frontend.passTLSClientCert.issuer=true
```
that would mimic this:
```
       Satisfy any
       SSLRequireSSL
       SSLVerifyClient require
       SSLVerifyDepth  10
       SSLOptions +ExportCertData +StrictRequire +LegacyDNStringFormat
       SSLOptions +StdEnvVars

       # Set SSL_* headers which are used by Django code
       RequestHeader set SSL_CLIENT_S_DN "%{SSL_CLIENT_S_DN}s"
       RequestHeader set SSL_CLIENT_I_DN "%{SSL_CLIENT_I_DN}s"
```
It's worth looking at the full range of [Apache support](https://httpd.apache.org/docs/current/mod/mod_ssl.html#envvars) for an understanding of what it does. You can see that they support both `SSL_CLIENT_S_DN` and `SSL_CLIENT_S_DN_$FIELD` headers.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

Issue templates help us help you by providing all necessary information.

Please edit your issue and use the available templates:

- [Feature request](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Feature_request.md)
- [Bug report](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Bug_report.md)

And remember: each time someone ignores the template, a cute little bunny dies.
<details>
<summary>#SaveTheCuteBunny :heart: :rabbit: :heart:</summary>

![southpark - estrella kill rabbit](https://user-images.githubusercontent.com/5674651/27143914-3cea7896-5130-11e7-81f8-02d8ac2da7a7.gif)

![southpark - estrella kill rabbit2](https://user-images.githubusercontent.com/5674651/27143924-49104c72-5130-11e7-82ae-ccedbd5dfa1c.gif)

</details>

**Comment by sirech** [comment]
a slightly related point: Should/Could traefik provide a way to whitelist certain CNs?

The use case I'm thinking about is as follows: You have Traefik in front of multiple microservices. Traefik is being accessed through mTLS from a known set of hosts. I would like to configure traefik so that it makes sure that only hosts with certain CN names in their certificate are allowed tot reach the services. Right now that check would have to be done per service, once you use the `passTLSClientCert`, which forces you to remember to do that for each new service that you spin up.

**Comment by traefiker** [comment]
Closed by #4298.
